compiler/src/iree/compiler/DispatchCreation/SetEncoding.cpp

```diff
@@ -120,168 +115,156 @@ static bool hasMatmulLikeBody(linalg::LinalgOp linalgOp) {
 }
 
 /// Not all contractions are supported by data tiling, so return true if:
-///   1) linalgOp has contraction indexingMaps.
-///   2) There are not more than one of each contraction dimension
-///   3) There is and M or N dimension, and there is a K dimension
-///   4) linalgOp has the same body as an ordinary int or float matmul
+///   1) linalgOp has pure tensor semantics.
+///   2) linalgOp does not have a preset compilation info.
+///   3) The workgroup count is not present if linalgOp is wrapped within
+///      Flow::DispatchRegionOp.
+///   4) All the operands do not have encodings.
+///   5) linalgOp has contraction indexingMaps.
+///   6) There are not more than one of each contraction dimension.
+///   7) There is and M or N dimension, and there is a K dimension.
+///   8) linalgOp has the same body as an ordinary int or float matmul.
 ///
 /// These restrictions are required because data tiling currently creates
 /// an Mmt4DOp or BatchMmt4DOp on the packed inputs.
 ///
 /// TODO(#16176): Loosen restrictions on contraction ops once data tiling
 /// can support more cases.
-static LogicalResult isSupportedContractionOp(PatternRewriter &rewriter,
-                                              linalg::LinalgOp linalgOp) {
+static bool isSupportedContractionOp(linalg::LinalgOp linalgOp) {
```

Comment:
This function will be moved to the new pass that attaches data-tiling hint attribute to operations.